src/Workspaces/CSharp/Portable/Utilities/CSharpUseImplicitTypeHelper.cs

```diff
@@ -259,9 +259,31 @@ protected override bool ShouldAnalyzeForEachStatement(ForEachStatementSyntax for
             }
 
             // variables declared using var cannot be used further in the same initialization expression.
-            if (initializer.DescendantNodesAndSelf()
-                    .Where(n => (n as IdentifierNameSyntax)?.Identifier.ValueText.Equals(identifier.ValueText) == true)
-                    .Any(n => semanticModel.GetSymbolInfo(n, cancellationToken).Symbol?.IsKind(SymbolKind.Local) == true))
+            bool isReusedInInitializer(SyntaxNode n)
+            {
+                // naming collision prerequisite
+                if (!(n is IdentifierNameSyntax id
+                    && id.Identifier.ValueText.Equals(identifier.ValueText)))
```

Comment:
IDE style is to put && on previous line.

Reply:
I would prefer this not change. No need to amend previous commit for this.

Reply:
> if we don't change it, then i would ask that we please indent things appropriately.

The indentation is correct at 4 spaces.

Reply:
> I had to stare at this a lot ...

The solution I thought of was this:

```csharp
if (!(n is IdentifierNameSyntax id)
    || !id.Identifier.ValueText.Equals(identifier.ValueText))
{
}
```

Reply:
> The indentation is always correct at 4 spaces (the only real exception in C# code is LINQ queries). Others are occasionally acceptable as well, but 4 spaces is never wrong.

Indentation of subexpressions within parenthesized expressions should align.  Otherwise you end up with code that is **super** confusing.  Is the subsequent boolean clause a sibling of the first preceding clause?  Or is it a sibling of a nested clause?  i.e.  when i see:

```c#
if (!(a...
    && b
```

my brain immediately starts pulling this apart as   ```!(a...```  as one clause and ```b``` as hte other.

But that's not hte case here.  ```&& b``` is associated with ```a``` and the ```!``` affects all of it. 

This is one of hte reasons i often push on people to not create overly complex boolean expressions as they end up hard to parse and hard to maintain over time.  To me, i would prefer something like:

```c#
if (!(n is ...
      id...)
```

Or, alternatively, just do:

```c#
var identifierNameMatches = n is IdentifierNameSyntax id && id.Identifier.ValueText.Equals(identifier.ValueText);
if (!identifierNameMatches)
{
...
```

Reply:
> The indentation is always correct at 4 spaces

Note: i disagree strongly with this.   And we've had enough bugs and confusion in Roslyn itself with code that follows this that i think we can apply sensible rules to complex expressions to aid in readability and maintainability.  

What we could say is: at a minimum indentation should be 4 spaces.  However, indentation should be applied to most bring the highest clarity to an expression. The more complex the expression, the more effort should be made to avoid potentially incorrect interpretations.

Reply:
@CyrusNajmabadi None of that is considered by the formatter that runs during typing in the IDE. Please feel free to file a feature request related to it, but no alignment-related item will become part of the code style checked during review prior to it being naturally supported by the IDE. Currently the only naturally-supported items in C# are:

* Alignment of LINQ clauses
* Indentation by 4 spaces relative to previous line

Reply:
> None of that is considered by the formatter that runs during typing in the IDE.

That's not true.  The formatter will not *format* in that manner.  But it will absolutely leave code alone that *is* formatted in that manner.  This is very intentional as attempting to codify all these potential relations has been out of scope of the formatter.  While, at the same time, it's been imperative to recognize that people *will* format in certain ways for clarity and the formatter should not undo that.

> but no alignment-related item will become part of the code style checked during review prior to it being naturally supported by the IDE or by an analyzer added to this repository

This is a pretty poor position to take.  It is completely reasonable to ask for things that help clarify the code and there are patterns and practices the IDE has followed for 20+ years.  Just ignoring that all out because we don't have a a mechanized way to enforce things is throwing the baby out with teh bathwater.

>  Currently the only naturally-supported items in C# 

Yes.  And it's fine for people to just use that approach by default.  However, in cases where code clarity is improved, it's utterly fine and reasonable to ask and to update the code.  Style/formatting are not *only* the thins that we have gotten the time to codify.  They're definitely things that we've institutionally adopted and which have a recognized value.  My position would be:

As long as it isn't onerous, we should err toward keeping the style consistent in hte IDE, and creating code that is readable, until the point that we can just have this enforced.  Otherwise, we just allow code readability to degrade in the meantime in areas that people aren't investing in.  

This does mean extra vigilance in code reviews.  But that's fine given the benefits to the health of the codebase (which shoudl be one of the highest priorities to the team, and one of the main goals of a PR in the first place).